### ISSUE TITLE
S4275 FP with private setter and collection deserialization

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#3441]
### Description

This scenario should be valid for deserialization purposes.

Coming from the community - https://community.sonarsource.com/t/s4275-getters-and-setters-should-access-the-expected-fields/24581/3

```
    public class ReproDeserializationScenario
    {
        private readonly List<int> _data = new List<int>();
        public List<int> Data
        {
            get => _data;
            // The setter in this case is only executed by a deserializer, but we don’t want to replace the field instance.
            private set // Noncompliant FP
            {
                foreach (var item in value)
                {
                    _data.Add(item);
                }
            }
        }
    }

```

### Known workarounds

Do the deserialization using the constructor.

### Related information

* C#/VB.NET Plugins Version 8.9

**Comment by andrei-epure-sonarsource** [comment]
note: this is a bit of an edge case, not sure it's worth adding complexity for the rule just for this

**Comment by miminno** [comment]
So what's the status on this one? #3443 suggests that it's been addressed, yet we're still getting tons of these false-positives across our projects.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1956](https://sonarsource.atlassian.net/browse/NET-1956)

[NET-1956]: https://sonarsource.atlassian.net/browse/NET-1956?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
